### PR TITLE
feat: wakuv2 store

### PR DESCRIPTION
**Pull request by richard-ramos**
Allows runnning a store node depending on node config settings.

### CODE REVIEW

**Review by alaibe — APPROVED** [review]

File: wakuv2/waku.go [hunk 1]
```diff
@@ -342,8 +352,11 @@ func (w *Waku) dnsDiscover(enrtreeAddress string, protocol libp2pproto.ID, apply
 	if !ok {
 		w.dnsAddressCacheLock.Lock()
 		var err error
-		multiaddresses, err = dnsdisc.RetrieveNodes(ctx, enrtreeAddress)
-		w.dnsAddressCache[enrtreeAddress] = multiaddresses
```
Comment (0x-r4bbit) [inline wakuv2/waku.go:346]: Was this a bug?

File: wakuv2/waku.go [hunk 2]
```diff
@@ -384,7 +397,7 @@ func (w *Waku) addWakuV2Peers(cfg *Config) {
 	}
 
 	addToStore := func(m multiaddr.Multiaddr, protocol libp2pproto.ID) {
-		peerID, err := w.node.AddPeer(m, protocol)
+		peerID, err := w.node.AddPeer(m, string(protocol))
```
Comment (0x-r4bbit) [inline wakuv2/waku.go:399]: Was there a breaking change such that now a string is required?

**Review by 0x-r4bbit — APPROVED** [review]
I hope I didn't miss any of the actual changes here (700+ files changed out of which +95% are changes in vendor)

File: wakuv2/waku.go (hunk 1 above)
Comment (richard-ramos) [reply wakuv2/waku.go:346]: I made changes in go-waku sometime ago due to a bug where I ignored that a ENR could be converted into multiple multiaddresses.

**Comment by richard-ramos** [comment]
> I hope I didn't miss any of the actual changes here (700+ files changed out of which +95% are changes in vendor)

:( indeed. The vendor folder did change a lot. FWIW, the actual changes are limited to these files

image

**Comment by qoqobolo** [comment]
Hi @richard-ramos,
sorry to disturb you but for some reason all builds (e2e, android, ios) are failing on the bundle stage in my status-mobile PR: https://github.com/status-im/status-mobile/pull/13809#issuecomment-1216735749
I tried to re-build them and use the script one more time, but it didn't help.
Logs for the e2e build: https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-13809/1/execution/node/66/log/

Could you take a look, please? If this issue is not on your side, I'll ping the infra team, however, I can see that builds from another go PR (and another developer) work fine on my side.

**Comment by richard-ramos** [comment]
> @qoqobolo: Could you take a look, please? If this issue is not on your side, I'll ping the infra team, however, I can see that builds from another go PR (and another developer) work fine on my side.

The issue is on my end. Working on a fix! 
Thank you!

**Comment by richard-ramos** [comment]
@qoqobolo, it should work now :) 

It might require a change to add the tag "gowaku_skip_migrations" like it was done here https://github.com/status-im/status-go/pull/2780/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R85 but i'm not sure

**Comment by richard-ramos** [comment]
:( now i'm getting a lot of lint issues locally, but at least jenkins passes the Lint stage

**Comment by qoqobolo** [comment]
@richard-ramos still doesn't work :(

https://ci.status.im/blue/organizations/jenkins/status-mobile%2Fprs%2Fandroid-e2e/detail/PR-13818/1/pipeline

Screenshot 2022-08-18 at 12 43 42

Screenshot 2022-08-18 at 12 44 44

**Comment by richard-ramos** [comment]
Hopefully now it works! :worried:

**Comment by qoqobolo** [comment]
> Hopefully now it works! 😟

Yaaay! It works now, launched tests. Thank you, Richard!

**Pull request merged**